### DISCUSSION TITLE
Can't upload videos longer than 30 seconds.

### DISCUSSION THREAD

**Discussion by merstoffe14** [tweepy/tweepy#2044] [Questions] [answered]
Please excuse me if this is not the correct place to post this question.

Using:
`media = api.media_upload(path) `

It often gives me the error that the video is longer than 30000 (ms) and that this is not allowed, even tough videos longer than 30 seconds are allowed on twitter.

**Comment by Harmon758** [comment]
This is not actionable, as there is no information here at all.
You need to provide what exactly you're trying and encountering.

**Comment by merstoffe14** [comment]
Excuse me, first time posting an issue to another repo.

I made this simple script to reproduce the error: 

```
import tweepy
from fileReader import FileReader
import sys

reader = FileReader("./accounts/"+sys.argv[1])

account_data = reader.read()

auth = tweepy.OAuthHandler(
                account_data['consumer_key'],
                account_data['consumer_secret']
                )
auth.set_access_token(
                account_data['access_token'],
                account_data['access_token_secret']
                )
api =tweepy.API(auth)

media = api.media_upload("test.mp4")

tweetinfo = api.update_status(status="test", media_ids=[media.media_id])
```

The filereader class is just to import the relevant api keys and tokens in to the script. After that I try to upload a file called 'test.mp4'. Which is a file I made 5 minutes ago. It is a video with a still picture that is 41 seconds long and 1 mb in size. I can tweet this video without any issues manually, but when I try to do it using this script it returns this error:

`tweepy.errors.BadRequest: 400 Bad Request
324 - Duration too long, maximum:30000, actual:41174 (MediaId: snf:1616970598772604930)`

**Comment by Harmon758** [comment, marked as answer]
You need to pass the `media_category` parameter for [`API.media_upload`](https://tweepy.readthedocs.io/en/v4.12.1/api.html#tweepy.API.media_upload) as `"tweet_video"`.

For code block usage, see https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.
